### DISCUSSION TITLE
Notification error - Discord

### DISCUSSION THREAD

**Discussion by Onslivion** [jef/streetmerchant#1267] [Help] [answered]
Just stumbled upon the issue with the recent Strix OC 3080 restock on Amazon. Only have discord notifications enabled, double checked webhooks and role mentions.

> [12:56:25 PM] error :: ✖ couldn't send discord message fields.flat is not a function {
  "stack": "TypeError: fields.flat is not a function\n    at Function.normalizeFields (/home/onslivion/ps5/node_modules/discord.js/src/structures/MessageEmbed.js:450:8)\n    at MessageEmbed.addFields (/home/onslivion/ps5/node_modules/discord.js/src/structures/MessageEmbed.js:266:42)\n    at MessageEmbed.addField (/home/onslivion/ps5/node_modules/discord.js/src/structures/MessageEmbed.js:257:17)\n    at /home/onslivion/ps5/build/notification/discord.js:33:23\n    at Object.sendDiscordMessage (/home/onslivion/ps5/build/notification/discord.js:60:11)\n    at Object.sendNotification (/home/onslivion/ps5/build/notification/notification.js:27:15)\n    at Object.<anonymous> (/home/onslivion/ps5/build/__test__/notification-test.js:25:16)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)"
}

**Comment by jef** [comment]
There is a new dependency. Use `npm install`.

**Comment by Onslivion** [comment]
Just pulled most recent build, did `npm install` and still resulting in same issue.

> npm WARN twilio@3.53.0 requires a peer of @types/express@^4.17.7 but none is installed. You must install peer dependencies yourself.
npm WARN twilio@3.53.0 requires a peer of @types/qs@6.9.4 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

None of these would be the issue, would they?

**Comment by jef** [comment]
Nah, shouldn't be.

**Comment by jef** [comment]
What does you .env look like?

**Comment by Onslivion** [comment]
ASCII_BANNER=""
ASCII_COLOR=""
AUTO_ADD_TO_CART=""
BROWSER_TRUSTED=""
COUNTRY=""
DESKTOP_NOTIFICATIONS=""
DISCORD_NOTIFY_GROUP="<@&785310895593095248>"
DISCORD_WEB_HOOK="[REDACTED]"
EMAIL_PASSWORD=""
EMAIL_TO=""
EMAIL_USERNAME=""
HEADLESS="true"
IN_STOCK_WAIT_TIME="120"
LOG_LEVEL=""
LOW_BANDWIDTH="true"
MAX_PRICE_SERIES_3060TI=""
MAX_PRICE_SERIES_3070=""
MAX_PRICE_SERIES_3080=""
MAX_PRICE_SERIES_3090=""
MAX_PRICE_SERIES_CORSAIR_SF=""
MAX_PRICE_SERIES_RX6800=""
MAX_PRICE_SERIES_RX6800XT=""
MAX_PRICE_SERIES_RX6900XT=""
MAX_PRICE_SERIES_RYZEN5600=""
MAX_PRICE_SERIES_RYZEN5800=""
MAX_PRICE_SERIES_RYZEN5900=""
MAX_PRICE_SERIES_RYZEN5950=""
MAX_PRICE_SERIES_SONYPS5C="550"
MAX_PRICE_SERIES_SONYPS5DE="550"
MAX_PRICE_SERIES_TEST=""
MICROCENTER_LOCATION=""
MQTT_BROKER_ADDRESS=""
MQTT_BROKER_PORT=""
MQTT_CLIENT_ID=""
MQTT_PASSWORD=""
MQTT_QOS=""
MQTT_TOPIC=""
MQTT_USERNAME=""
NVIDIA_ADD_TO_CART_ATTEMPTS=""
NVIDIA_SESSION_TTL=""
OPEN_BROWSER="false"
PAGE_BACKOFF_MIN="32000"
PAGE_BACKOFF_MAX="900000000000000"
PAGE_SLEEP_MIN="5000"
PAGE_SLEEP_MAX="8750"
PAGE_TIMEOUT=""
PAGERDUTY_INTEGRATION_KEY=""
PAGERDUTY_SEVERITY=""
PHILIPS_HUE_API_KEY=""
PHILIPS_HUE_CLOUD_ACCESS_TOKEN=""
PHILIPS_HUE_CLOUD_CLIENT_ID=""
PHILIPS_HUE_CLOUD_CLIENT_SECRET=""
PHILIPS_HUE_CLOUD_REFRESH_TOKEN=""
PHILIPS_HUE_LAN_BRIDGE_IP=""
PHILIPS_HUE_LIGHT_COLOR=""
PHILIPS_HUE_LIGHT_IDS=""
PHILIPS_HUE_LIGHT_PATTERN=""
PHONE_CARRIER=""
PHONE_NUMBER=""
PLAY_SOUND=""
PROXY_ADDRESS=""
PROXY_PROTOCOL=""
PROXY_PORT=""
PUSHBULLET=""
PUSHOVER_TOKEN=""
PUSHOVER_USER=""
PUSHOVER_PRIORITY=""
SCREENSHOT=""
SHOW_ONLY_BRANDS=""
SHOW_ONLY_MODELS=""
SHOW_ONLY_SERIES="sonyps5c,sonyps5de"
SLACK_CHANNEL=""
SLACK_TOKEN=""
SMTP_ADDRESS=""
SMTP_PORT=""
STORES="adorama,amazon,bestbuy,bandh,gamestop,target,walmart,playstation,newegg"
TELEGRAM_ACCESS_TOKEN=""
TELEGRAM_CHAT_ID=""
TWILIO_ACCOUNT_SID=""
TWILIO_AUTH_TOKEN=""
TWILIO_FROM_NUMBER=""
TWILIO_TO_NUMBER=""
TWITCH_ACCESS_TOKEN=""
TWITCH_CHANNEL=""
TWITCH_CLIENT_ID=""
TWITCH_CLIENT_SECRET=""
TWITCH_REFRESH_TOKEN=""
TWITTER_ACCESS_TOKEN_KEY=""
TWITTER_ACCESS_TOKEN_SECRET=""
TWITTER_CONSUMER_KEY=""
TWITTER_CONSUMER_SECRET=""
TWITTER_TWEET_TAGS=""
USER_AGENT=""
WEB_PORT=""

**Comment by JeremieLeblanc** [comment, marked as answer]
I believe updating node should fix this issue.

**Comment by Onslivion** [comment]
nodejs version is v10.19.0.

**Comment by JeremieLeblanc** [comment]
Array.flat was only added to node v11
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat#Browser_compatibility

**Comment by Onslivion** [comment]
Forcing an update using the following solution for Ubuntu 20.04 resolved the issue. Thank you.

> sudo npm cache clean -f
> sudo npm install -g n
> sudo n stable
